RFC-0036-hardware-accelerators-pytorch.org.md

```diff
@@ -0,0 +1,89 @@
+
+
+## **Summary**
+The main page of pytorch.org only shows PyTorch Get-Started  on CUDA, ROCm and CPU, but the hardware accelerator support is more diverse and the ecosystem is rapidly evolving with new accelerators like Intel Gaudi, Google TPU, etc.
+The proposal in this RFC is to enable the visibility of broad set of accelerators that supports PyTorch for the community, and create a space for hardware vendors to provide instructions and details about their PyTorch support.  
+
+
+## **Motivation**
+There are 2 primary motivations behind this RFC:
+1. Evangelize PyTorch as an open source framework that supports a wide variety of hardware.
+   * PyTorch is supported on more compute platforms than CUDA, ROCm and CPU .
+   * PyTorch can maintain and enhance its competitive position in the machine learning frameworks  by enabling the visibility of rich set of hardware accelerators on which PyTorch can run.
+2. Provide a consolidated entry point where to look for more information about supported hardware accelerators.
+   * pytorch.org is a starting point for many users. The page can help users to find more details.
+
+
+
+## **Proposed Implementation**
+The proposal is to enhance pytorch.org by adding 3  new sections:
+* Add new compute platforms to “Install PyTorch” section at the main page of pytorch.org.
+* Add new compute platforms to https://pytorch.org/get-started/.
+* Add new “Compute Platforms” section at the main page of pytorch.org.
+
+### PyTorch integration types
+There are at least 2 ways how compute platforms are integrated with PyTorch:
+1. In-tree – CPU, CUDA, ROCm are developed, built and tested in PyTorch environment. PyTorch is ensuring quality criteria. This approach is limited to only a few compute platforms and it does not scale with number of compute platforms.
+2. Out-of-tree – Integration of other compute platforms like Intel Gaudi is done via additional python package (extension) that needs to be installed on top of PyTorch CPU package. Development, built and testing is done outside of PyTorch. In this case:
+   * PyTorch is ensuring quality of PyTorch CPU package.
+   * Hardware provider is ensuring quality of its extension against PyTorch CPU package.
+
+### Stable vs. nightly builds
+1. Stable builds – the hardware provider shall provide the installation commands when a compute platform is tested against given PyTorch version and it meets the quality criteria set by PyTorch Foundation.    
+2. Preview (nightly) -  Similar to the stable builds, but the hardware provider must implement method to provide quickly fixes for PyTorch nightly.   It should be optional.
+
+### Quality criteria
+PyTorch foundation shall introduce minimal requirements for new compute platforms.
+Report what kind of testing was done on a compute platform with PyTorch build
+* A common test report format - to be defined.
+  * A test report to contain: results of tests (PyTorch UT, TorchBench, model runs, additional test suites), tested OSes.
```

Comment:
Due to variations in capabilities and implementation among different hardware accelerators, how should the minimum testing suite be defined?

Reply:
This is a challenge. One option is to verify basic infrastructure and ops, i.e. tensor management, dispatcher, copy ops, basic math, hello world, etc.
The device capabilities reporting in PyTorch is not very extensive and extending it to take care of all hardware variants would be overengineering.